### PR TITLE
Do not check if the Lifecycle argument is null in DefaultBinding

### LINKED ISSUE #607: DefaultBinding with null endpoint fails after AbstractMessageChannelBinder.doBindProducer()
While implementing a SCS binder, dealing with MessageHandler in the producer side of it, if it does not implement the interface Lifecycle, `AbstractMessageChannelBinder` will send a null reference to `DefaultBinding`, which will cause an assertion exception, since the constructor verifies it.

```
public DefaultBinding(String name, String group, T target, Lifecycle endpoint) {
        Assert.notNull(target, "target must not be null");
        Assert.notNull(endpoint, "endpoint must not be null");
        this.name = name;
        this.group = group;
        this.target = target;
        this.endpoint = endpoint;
    }
```

The result is that non lifecycle implementing message handlers as the JmsSendingMessageHandler will not work out of the box.

### PR DESCRIPTION
**Pull request by mbogoevici**
Fixes #607

Signed-off-by: Marius Bogoevici <mbogoevici@pivotal.io>

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java [hunk 1]
```diff
@@ -38,15 +38,22 @@
 
 	protected final T target;
 
-	protected final Lifecycle endpoint;
-
-	public DefaultBinding(String name, String group, T target, Lifecycle endpoint) {
+	protected final Lifecycle runningComponent;
+
+	/**
+	 * Creates a new instance.
```
Comment (ilayaperumalg) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: Can be more descriptive than just `Creates a new instance`.

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java [hunk 2]
```diff
@@ -38,15 +38,22 @@
...
+	 *                         and will be stopped during unbinding
+	 */
+	public DefaultBinding(String name, String group, T target, Lifecycle runningComponent) {
 		Assert.notNull(target, "target must not be null");
-		Assert.notNull(endpoint, "endpoint must not be null");
 		this.name = name;
 		this.group = group;
 		this.target = target;
-		this.endpoint = endpoint;
+		this.runningComponent = runningComponent;
```
Comment (ilayaperumalg) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: While `runningComponent` is a good name in general here, we could as well think of some other name as it is not yet running here. May be just `lifecycleComponent` ?

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java (hunk 1 above)
Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: Do you have any suggestion in particular?

Comment (ilayaperumalg) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: I was thinking something simple description like `Creates an instance that contains information about the binding of the producer/consumer target`

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 8e5689c

**Pull request closed without merging**